variables.tf

```diff
@@ -242,6 +242,10 @@ variable "efs_configs" {
     file_system_id = string
     root_directory = string
     container_path = string
+    authorization_config = object({
+      access_point_id = string
+      iam = string
```

Comment:
should be able to set the type to *null*
https://www.terraform.io/language/expressions/types#null

In the end both conditions need to be met:
- deploy with *authorization_config*
- deploy without *authorization_config*